src/modules/plus/filter_strobe.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * filter_strobe.c -- simple strobing filter
+ * Copyright (C) 2007 Martin Sandsmark <martin.sandsmark@kde.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt_filter.h>
+#include <framework/mlt_frame.h>
+#include <framework/mlt_producer.h>
+#include <framework/mlt_service.h>
+#include <framework/mlt_factory.h>
+#include <framework/mlt_property.h>
+
+#include <stdio.h>
+#include <string.h>
+
+static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	// Get the image
+	mlt_filter filter = mlt_frame_pop_service( frame );
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+
+	int error = mlt_frame_get_image( frame, image, format, width, height, 1 );
+	if (error) {
+		return error;
+	}
+
+	int invert = mlt_properties_get_int( properties, "strobe_invert" );
+	int interval = mlt_properties_get_int( properties, "interval" );
```

Comment:
Do you want to make the interval keyframable? It is fairly easy but not required.

Reply:
I thought about it, but didn't really see the point for myself.

Reply:
Do I just need to set `mutable: yes` and switch to `mlt_properties_anim_get_int`?

Reply:
You may not see the point, but this will get exposed to users probably through Kdenlive who will then ask for it repeatedly.

`mutable: yes` is not for animation; it simply means when the app changes the value the filter sees the new value instead of only the initial. We do not yet have a flag in service metadata for animation keyframes.

`mlt_properties_anim_get_int` mostly, yes, but that function needs a position and duration:
```
	mlt_position position = mlt_filter_get_position( filter, frame );
	mlt_position length = mlt_filter_get_length2( filter, frame );
```

Reply:
> You may not see the point, but this will get exposed to users probably through Kdenlive who will then ask for it repeatedly.

Yup, that's why I want to implement it, was just explaining why it wasn't done initially.